tools/rex-asset.c

```diff
@@ -47,30 +56,74 @@ int main (int argc, char **argv)
     struct rex_header header = rex_create_header();
     rex_write_header (fp, &header);
 
-    // write texture file
-    FILE *t = fopen (argv[1], "rb");
-    if (!t)
-        die ("Cannot open rex asset file %s\n", argv[1]);
-    fseek (t, 0, SEEK_END);
-    uint64_t sz = ftell (t);
-    fseek (t, 0, SEEK_SET);
-    uint8_t *blob = malloc (sz);
-    if (fread (blob, sz, 1, t) != 1)
-        die ("Cannot read rexasset content");
-    fclose (t);
-
-    if (rex_write_rexasset_bock (fp, &header, blob, sz, "rexasset", 0))
+    char *file_ending = ".a_rexasset";
+    if (write_asset_package_to_rex_file(fp, &header, argv[1], file_ending, 0))
     {
-        warn ("Error during file write %d\n", errno);
-        free (blob);
         fclose (fp);
-        return -1;
+        die ("Could not write asset package %s%s to REX file\n", argv[1], file_ending);
+    }
+    file_ending = ".i_rexasset";
+    if (write_asset_package_to_rex_file(fp, &header, argv[1], file_ending, 1))
+    {
+        fclose (fp);
+        die ("Could not write asset package %s%s to REX file\n", argv[1], file_ending);
+    }
+    file_ending = ".w_rexasset";
+    if (write_asset_package_to_rex_file(fp, &header, argv[1], file_ending, 2))
+    {
+        fclose (fp);
+        die ("Could not write asset package %s%s to REX file\n", argv[1], file_ending);
     }
 
     rex_write_header (fp, &header);
 
     fclose (fp);
-    free (blob);
     return 0;
 }
 
+int write_asset_package_to_rex_file(FILE *fp, struct rex_header *header, char *file_name, char *file_ending, uint64_t id)
+{
+    // write asset file
+    char *file_name_with_ending;
+    file_name_with_ending = malloc(strlen(file_name) + strlen(file_ending) + 1);
+    strcpy(file_name_with_ending, file_name);
+    strcat(file_name_with_ending, file_ending);
+
+    FILE *asset_file = fopen (file_name_with_ending, "rb");
+    if (!asset_file)
+        die ("Cannot open rex asset file %s\n", file_name_with_ending);
+    fseek (asset_file, 0, SEEK_END);
+    uint64_t file_size = ftell (asset_file);
+    fseek (asset_file, 0, SEEK_SET);
+    uint8_t *blob = malloc (file_size);
+    if (fread (blob, file_size, 1, asset_file) != 1)
+        die ("Cannot read content of %s\n", file_name_with_ending);
+    fclose (asset_file);
+
+    uint16_t target_platform;
+    if(strcmp(file_ending,".a_rexasset") == 0)
+    {
```

Comment:
no curlies for onliners for our c style. Please use the following astyle config for openrex:


```
--style=allman --indent=spaces=4 --align-pointer=name --align-reference=name --indent-switches --indent-cases --pad-oper --pad-paren-out --pad-header --unpad-paren --indent-namespaces --remove-braces --convert-tabs --mode=c
```